hircluster.c

```diff
@@ -2257,29 +2230,26 @@ static void *redis_cluster_command_execute(redisClusterContext *cc,
 
     node = node_get_by_table(cc, (uint32_t)command->slot_num);
     if (node == NULL) {
-        __redisClusterSetError(cc, REDIS_ERR_OTHER, "node get by table error");
         goto error;
     }
 
     c = ctx_get_by_node(cc, node);
-    if (c == NULL) {
-        goto error;
-    } else if (c->err) {
-        node = node_get_which_connected(cc);
-        if (node == NULL) {
-            __redisClusterSetError(cc, REDIS_ERR_OTHER,
-                                   "no reachable node in cluster");
+    if (c == NULL || c->err) {
+        /* Failed to connect. Maybe there was a failover and this node is gone.
+         * Update slotmap to find out. */
+        if (cluster_update_route(cc) != REDIS_OK) {
             goto error;
         }
 
-        cc->retry_count++;
-        if (cc->retry_count > cc->max_retry_count) {
-            __redisClusterSetError(cc, REDIS_ERR_CLUSTER_TOO_MANY_RETRIES,
-                                   "too many cluster retries");
+        redisClusterNode *old_node = node;
+        node = node_get_by_table(cc, (uint32_t)command->slot_num);
+        if (node == NULL) {
             goto error;
         }
+        if (node != old_node) {
```

Comment:
I believe this would always be true after a cluster_update_route() so the check can be skipped.
The node objects are always replaced during a slotmap update, but the hiredis context node->con is kept (see `cluster_nodes_swap_ctx()`).

Reply:
Interesting. So instead we can save old_ctx och jämföra `c != old_ctx` istället?